Azure/sonic-buildimage
# Update process-reboot-cause to save the reboot-cause to state-db

**- Why I did it**
To make the reboot cause history information available for telemetry service, during processing the reboot-cause on boot, the information is saved in state-db.
**- How I did it**
Split the `process-reboot-cause service` into two services: `determine-reboot-cause` and `process-reboot-cause`
1. determine-reboot-cause :
 Determine the reboot-cause based on the hardware/software reboot-cause information and save it into a file with time stamp when it's determined.
2. process-reboot-cause : 
 Read the saved reboot-cause files up to 10 files (after time-sorted) from "/host/reboot-cause/history" directory and save them to state-db. 
**- How to verify it**
Tested in DUT using Cli  commands ( introduced and modified with this PR : https://github.com/Azure/sonic-utilities/pull/1154) and check the state-db with redis-cli.

**- Which release branch to backport (provide reason below if selected)**




**- Description for the changelog**

HLD : https://github.com/Azure/SONiC/pull/669

**- A picture of a cute animal (not mandatory but encouraged)**
